### PR TITLE
Add support to desktop environments

### PR DESCRIPTION
**Pull request by javiersantossanchez**

### CODE REVIEW

File: pengwin-setup.d/desktop.sh [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+
+function install_dependencies() {
+  local dependencies_instaled
+  echo "installing dependencies"
+  bash "${SetupDir}"/guilib.sh --yes "$@"
+  if [[ -f /etc/profile.d/dbus.sh ]]; then
+    bash "${SetupDir}"/hidpi.sh --yes "$@"
+    if [[ -f /etc/profile.d/hidpi.sh ]]; then
+        dependencies_instaled=0
```
Comment (crramirez) [inline pengwin-setup.d/desktop.sh:44]: Correct the indent to two spaces

File: pengwin-setup.d/desktop.sh [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    return 1
+  else
+    return 0
+  fi
+}
+
+
+function install_dependencies() {
+  local dependencies_instaled
+  echo "installing dependencies"
```
Comment (crramirez) [inline pengwin-setup.d/desktop.sh:39]: Capitalize "Installing"

File: pengwin-setup.d/desktop.sh [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+function install_xrdp() {
+  install_packages xrdp
+  install_packages xorgxrdp
+  sudo sed -i 's/3389/3390/g' /etc/xrdp/xrdp.ini
+  sudo /etc/init.d/xrdp start
+  
+  sudo bash -c 'cat > /usr/local/bin/remote_desktop.sh' << EOF
+#!/bin/bash
+
+function execute_remote_desktop(){
```
Comment (crramirez) [inline pengwin-setup.d/desktop.sh:65]: Add a space before the {

File: pengwin-setup.d/desktop.sh [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    host_ip=\$(ip -o -f inet addr show | grep -v 127.0.0 | awk '{printf "%s", \$4}' | cut -f1 -d/)
```
Comment (crramirez) [inline pengwin-setup.d/desktop.sh:66]: Indent 2 spaces

File: pengwin-setup.d/desktop.sh [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+  return $dependencies_instaled
+}
+
+function install_xrdp() {
+  install_packages xrdp
+  install_packages xorgxrdp
+  sudo sed -i 's/3389/3390/g' /etc/xrdp/xrdp.ini
+  sudo /etc/init.d/xrdp start
+  
+  sudo bash -c 'cat > /usr/local/bin/remote_desktop.sh' << EOF
```
Comment (crramirez) [inline pengwin-setup.d/desktop.sh:62]: Use `sudo tee` like in the line 82

File: pengwin-setup.d/desktop.sh [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+    user_name=\$(whoami)
+    echo "username:s:\$user_name" > /tmp/remote_desktop_config.rdp
+    cd /tmp
+    mstsc.exe remote_desktop_config.rdp  /v:\$host_ip:3390 /f
+}
+
+execute_remote_desktop
+EOF
+
+    sudo bash -c 'cat > /usr/local/bin/start-xrdp' << EOF
```
Comment (crramirez) [inline pengwin-setup.d/desktop.sh:76]: Use `sudo tee` like in the line 82

File: pengwin-setup.d/desktop.sh [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+  fi
+}
+
+
+function install_dependencies() {
+  local dependencies_instaled
+  echo "installing dependencies"
+  bash "${SetupDir}"/guilib.sh --yes "$@"
+  if [[ -f /etc/profile.d/dbus.sh ]]; then
+    bash "${SetupDir}"/hidpi.sh --yes "$@"
```
Comment (crramirez) [inline pengwin-setup.d/desktop.sh:42]: Add `--quiet` to avoid the last message

File: pengwin-setup.d/desktop.sh [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+
+function install_xfce() {
+  if install_dependencies "$@" ; then
+    install_packages xfce4-terminal
+    install_packages xfce4
+    install_xrdp
+    if package_installed "xfce4-terminal" && package_installed "xfce4"; then
+      create_shortcut "Xfce desktop (WSL)" "/usr/local/bin/remote_desktop.sh" "/usr/share/pixmaps/xfce4_xicon.png"
+    else
+      echo "There is a problem with xfce4 instalation"
```
Comment (crramirez) [inline pengwin-setup.d/desktop.sh:103]: There is a problem with xfce4 **installation**

File: pengwin-setup.d/gui.sh [hunk 9]
```diff
@@ -12,6 +12,7 @@ function main() {
   local menu_choice=$(
 
     menu --title "GUI Menu" --checklist --separate-output "Install an X server or various other GUI applications\n[SPACE to select, ENTER to confirm]:" 17 99 9 \
+      "DESKTOP" "Install Desktops enviroments" off \
```
Comment (crramirez) [inline pengwin-setup.d/gui.sh:15]: Install **Desktop** **environments**

File: pengwin-setup.d/desktop.sh [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+    fi
+  fi
+
+}
+
+function main() {
+  local menu_choice=$(
+
+    menu --title "GUI Menu" --checklist --separate-output "Install an X server or various other GUI applications\n[SPACE to select, ENTER to confirm]:" 8 50 2 \
```
Comment (crramirez) [inline pengwin-setup.d/desktop.sh:112]: `menu --title "Desktop Menu" --checklist --separate-output "Install Desktop environments\n[SPACE to select, ENTER to confirm]:" 10 55 1 \`

File: pengwin-setup.d/desktop.sh [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+      "XFCE" "Install XFCE Desktop environment" off \
```
Comment (crramirez) [inline pengwin-setup.d/desktop.sh:113]: `"XFCE" "Install XFCE Desktop environment " on`

**Review by crramirez — CHANGES_REQUESTED** [review]
Some minor changes to improve the workflow and code style

File: pengwin-setup.d/desktop.sh (hunk 2 above)
Comment (javiersantossanchez) [reply pengwin-setup.d/desktop.sh:38]: Done

File: pengwin-setup.d/desktop.sh (hunk 3 above)
Comment (javiersantossanchez) [reply pengwin-setup.d/desktop.sh:80]: Done

File: pengwin-setup.d/desktop.sh (hunk 8 above)
Comment (javiersantossanchez) [reply pengwin-setup.d/desktop.sh:124]: Done

File: pengwin-setup.d/gui.sh (hunk 9 above)
Comment (javiersantossanchez) [reply pengwin-setup.d/gui.sh:15]: Done

File: pengwin-setup.d/desktop.sh (hunk 10 above)
Comment (javiersantossanchez) [reply pengwin-setup.d/desktop.sh:133]: Done

File: pengwin-setup.d/desktop.sh (hunk 11 above)
Comment (javiersantossanchez) [reply pengwin-setup.d/desktop.sh]: Done

File: pengwin-setup.d/desktop.sh (hunk 7 above)
Comment (javiersantossanchez) [reply pengwin-setup.d/desktop.sh]: Done

**Review by crramirez — APPROVED** [review]

**Pull request merged**